jupyter_notebooks/Book_1_Intro_Williamson_2.ipynb

```diff
@@ -0,0 +1,380 @@
+   "id": "1e1ae651",
+   "metadata": {},
+   "source": [
+    "We can specify which diagnostics we wish to record over a simulation. The list of avaliable diagnostics can be found in the gusto source code: https://github.com/firedrakeproject/gusto/blob/main/gusto/diagnostics.py \\\n",
+    "We pass these diagnostics into the State function. This will initialise all the necessary components of the test case. Gusto will update this state after each iteration of the chosen time-stepper."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "db9d61a0",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "gusto:INFO Physical parameters that take non-default values:\n",
+      "gusto:INFO H: 5960.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "diagnostic_fields = [RelativeVorticity(), PotentialVorticity(),\n",
+    "                         ShallowWaterKineticEnergy(),\n",
+    "                         ShallowWaterPotentialEnergy(),\n",
+    "                         ShallowWaterPotentialEnstrophy(),\n",
+    "                        CourantNumber()]\n",
+    "\n",
+    "state = State(mesh,\n",
+    "              dt=dt,\n",
+    "              output=output,\n",
+    "              parameters=parameters,\n",
+    "              diagnostic_fields=diagnostic_fields)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "80f8c00a",
+   "metadata": {},
+   "source": [
+    "A spatially varying Coriolis force is defined over the sphere. We then can set-up our form of the shallow water equations by passing the state and Coriolis parameter into the ShallowWaterEquations() function. We additionally specify the function space (BDM) and degree (1) that we want to use with the Compatiable Finite Element method. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "8ed9914e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Create a spatially varying function for the Coriolis force:\n",
+    "Omega = parameters.Omega\n",
+    "x = SpatialCoordinate(mesh)\n",
+    "fexpr = 2*Omega*x[2]/R\n",
+    "eqns = ShallowWaterEquations(state, \"BDM\", 1, fexpr=fexpr)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9918a45e",
+   "metadata": {},
+   "source": [
+    "We are now ready to specify the initial conditions.\n",
+    "Due to our choice of function spaces for the velocity and depth, the intialisations of each variable use projection \n",
+    "and interpolation operations respectively. Williamson 2 specifies initial conditions for a 'Global Steady State Non-linear Zonal Geostrophic Zone' (do I need this detail?). \n",
+    "$$ \\textbf{u}_0 = \\frac{u_{max}}{R} [-y,x,0] $$\n",
+    "$$ h_0 = H - \\frac{\\Omega u_{max} z^2}{g R}  $$\n",
+    "We only require a two-dimensional velocity, so the vertical component is specified to be zero. (Jemma/Tom question: Why is $R/R^2$ used in the code)???"
```

Comment:
I think we are embedded in a 3D space so we are specifying all three of the Cartesian components of the velocity. I don't think we need the `R/R^2` so maybe you can cancel one out! Presumably the was originally done to make it clear that the final part is `sin^2(lat)`?